Tasks/Gradle/sonarqube-common/sonarqube-qualitygate.ts

```diff
@@ -0,0 +1,42 @@
+/// <reference path="../../../definitions/vsts-task-lib.d.ts" />
+
+import tl = require('../testlib/tasklib-wrapper');
+
+// Creates a build summary section to display the quality gate status.
+// Returns null if analysisStatus was null.
+export function createBuildSummaryQualityGateSection(analysisStatus:string):string {
```

Comment:
The HTML code, colours, etc. used in this class were mostly directly ported from the Powershell implementation for MSBuild SonarQube integration, to keep the experience consistent.